presto-main/src/main/java/io/prestosql/operator/scalar/FormatNumberFunction.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.StandardTypes;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.lang.String.format;
+
+public final class FormatNumberFunction
+{
+    private FormatNumberFunction() {}
+
+    @Description("formats large number using a unit symbol")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice formatNumber(@SqlType(StandardTypes.BIGINT) long num)
+    {
+        return utf8Slice(formatCount(num));
+    }
+
+    @Description("formats large number using a unit symbol")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice formatNumber(@SqlType(StandardTypes.DOUBLE) double num)
+    {
+        return utf8Slice(formatCount((long) num));
+    }
+
+    private static String formatCount(long count)
+    {
+        double fractional = count;
+        String unit = "";
+        if (fractional >= 1000 || fractional <= -1000) {
+            fractional /= 1000;
+            unit = "K";
+        }
+        if (fractional >= 1000 || fractional <= -1000) {
+            fractional /= 1000;
+            unit = "M";
+        }
+        if (fractional >= 1000 || fractional <= -1000) {
+            fractional /= 1000;
+            unit = "B";
+        }
+        if (fractional >= 1000 || fractional <= -1000) {
+            fractional /= 1000;
+            unit = "T";
+        }
+        if (fractional >= 1000 || fractional <= -1000) {
+            fractional /= 1000;
+            unit = "Q";
+        }
+
+        return format("%s%s", getFormat(fractional).format(fractional), unit);
```

Comment:
This could just be `getFormat(fractional).format(fractional) + unit`. No need to call `format()`